### ISSUE TITLE
Request: adding keyframe timeline so its easier to edit already created 

### ISSUE THREAD

**Issue by timmermansjoy** [opencv/cvat#5003]
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

When reviewing a video clip that has been created with tracks, its hard to see where keyframes are created.

### Expected Behaviour

Having the ability to see a timeline with all the keyframes that have been made over time. So editing and merging annotations are easier. As seen in an example here: 

![Screenshot 2022-09-27 at 12 38 49 PM](https://user-images.githubusercontent.com/61321383/192505039-8eec48d9-07d8-483b-9a1d-9618f0fe30df.png)

### Current Behaviour

only see a timeline that shows the frame count

### Possible Solution

Adding a timeline like shown in the example image. 

### Context

When overviewing a labeling task it is hard to see where keyframes have been made when you want to edit a keyframe that has been made in the past

**Comment by zhiltsov-max** [comment]
Duplicate of #1228, #1796

**Comment by titarch** [comment]
Another suggestion: a basic implementation I made using d3, that can show labels or attributes, all zoomable and pannable and with a frame cursor.

![image](https://user-images.githubusercontent.com/23527994/223978114-b0f2e098-82cd-434b-b1a9-df5b9f5d4a8f.png)

**Comment by potipot** [comment]
@titarch would you be able to share some more information on how you created such timeline view? It would be super useful for my use case where attribute changes over time

**Comment by Stijnp** [comment]
Also interested in a feature like this. The open source [VGG Image Annotator](https://github.com/ox-vgg/via/tree/master) has something that could potentially be used as a starting point. https://www.robots.ox.ac.uk/~vgg/software/via/demo/via_video_annotator.html

**Comment by bparsy-odin** [comment]
@potipot Here are some details about the feature:

The job timeline is an interactive component meant for viewing, navigat-
ing and editing the annotations of a job. It is similar in functionality to the
timeline of a video editing or animation software. In the context of CVAT:
• The x-axis represents the frame number
• On the y-axis, each track is represented by a line
• In the middle we added low resolution thumbnails of the frames (we have added this later, it is not part of the screenshot I shared)
• Normal tracks (i.e. not tags) are above the thumbnails
• Tag tracks (image attributes) are below the thumbnails (depends on issue #5716)
• The timeline is zoomable and pannable using the mouse wheel and mouse drag respectively
• Holding ctrl sticks the timeline cursor to the mouse and clicking on two frames creates a new track between them using the selected label in timeline controls
• The timeline shows the buffered and saved frames at the bottom

Regarding implementation, most of it is contained in a (relatively simple) single component using a d3 chart to draw from the directly annotation's data and metadata.

**Comment by potipot** [comment]
@bparsy-odin Thank you for the provided details. They will be very helpful
